WordPress/src/main/java/org/wordpress/android/ui/JetpackConnectionDeeplinkActivity.java

```diff
@@ -91,48 +88,35 @@ protected void onCreate(Bundle savedInstanceState) {
     }
 
     @Override
-    protected void onStop() {
-        mDispatcher.unregister(this);
-        super.onStop();
+    protected void onActivityResult(int requestCode, int resultCode, Intent data) {
+        super.onActivityResult(requestCode, resultCode, data);
+        if (requestCode == RequestCodes.JETPACK_LOGIN && resultCode == RESULT_OK) {
+            trackResult();
+        }
+        finishAndGoBackToSource();
     }
 
     @Override
-    protected void onStart() {
-        super.onStart();
-        mDispatcher.register(this);
+    public void onBackPressed() {
+        super.onBackPressed();
+        finishAndGoBackToSource();
     }
 
-    @Override
-    protected void onActivityResult(int requestCode, int resultCode, Intent data) {
-        super.onActivityResult(requestCode, resultCode, data);
-        if (requestCode == RequestCodes.JETPACK_LOGIN && resultCode == RESULT_OK && mSource == Source.STATS) {
-            showStats();
+    private void trackResult() {
+        if (!TextUtils.isEmpty(mReason)) {
+            AppLog.e(AppLog.T.API, "Could not connect to Jetpack, reason: " + mReason);
+            ToastUtils.showToast(this, mReason);
         } else {
-            finish();
+            trackWithSource(Stat.SIGNED_INTO_JETPACK, mSource);
         }
     }
 
-    private void showStats() {
-        if (!TextUtils.isEmpty(mReason)) {
-            ToastUtils.showToast(this, mReason);
-            finish();
-        } else {
+    private void finishAndGoBackToSource() {
+        if (mSource == Source.STATS) {
             SiteModel site = (SiteModel) getIntent().getSerializableExtra(SITE);
             mDispatcher.dispatch(SiteActionBuilder.newFetchSiteAction(site));
```

Comment:
Can you let me know your thoughts on what this dispatch do? I am trying to understand the addition of `StatsActivity.LOGGED_INTO_JETPACK` which I think is related to that. To understand what should happen and when, I'll have to think about this myself, but I'd love to hear your thoughts to better understand the said change.

Reply:
So how I understand it - We get back the callback that the site is connected to Jetpack. The local SiteModel (and Store) is not aware of this and still say that it is not connected. I wanted to trigger a page refresh to get the current page state in order to show Stats. What do you think?

Reply:
@planarvoid I _think_ this might be good enough for now, I am not 100% sure honestly. I have opened #7473 to separate this issue. The one thing that'd really help me to figure this out (or at least leave information to the person who will) is some testing instructions. Can you let me know some steps to reproduce the need for this?

Reply:
If this is disabled, after logging into Jetpack the user is redirected to Stats but the Stats are empty because the app thinks Jetpack is not connected and it doesn't automatically sync the site. The resulting Stats page looks like this (without the pull to refresh progress bar).

![screenshot_1520869150](https://user-images.githubusercontent.com/1079756/37293595-0e99cbc2-2614-11e8-969c-132aabb999f1.png)

Reply:
I understand that the site need to be fetched, but when a login occurs we fetch all the sites for that account. If the login occurs after the Jetpack connection is completed, we shouldn't really need this.

My question was not about whether we need this or not, it was more about what's the exact scenario we need this in, so we can better handle that in the stats page. Does that make sense?

Reply:
Login usually happens before the Jetpack connection is completed because it is part of the flow (you log in, you approve Jetpack connection). The login in the Deeplink activity only happens when you're are logged in in the WebView and not in the app.

I honestly think the best way would be to rewrite the Stats page (and I have from the beginning) but it felt out of scope 👍

Reply:
In this case, when we fetch the site, do we actually fetch it from the REST API or the xml-rpc? Because it doesn't look like we make any changes to the current site, which means as far as FluxC knows, this is still a self-hosted site and dispatching the fetch action will not actually pull the Jetpack details. It looks like the flow is currently working, so there is something going on that I don't know about.

I am just leaving this note here, so I can take a look when/if I work on that issue. Unfortunately, this issue will probably take more time than it's worth to us right now, so it's not a priority.